cypress/integration/site-search/index.spec.ts

```diff
@@ -0,0 +1,95 @@
+const NO_RESULT = 'no_result';
+
+const addQueryMetaData = ({query, result}) => {
+  const [results] = result.results;
+
+  return {results: [{...results, query, params: `query=${query}&hitsPerPage=5`}]};
+};
+const extractRequestParams = (parsedBody) => Cypress._.get(parsedBody, 'requests[0].params');
+const extractHits = (xhr) => Cypress._.get(xhr, 'response.body.results[0].hits', []);
+
+const testSearch = (searchPhrase) => {
```

Comment:
Most of this test section seems to cover the majority of what we want to expect, perfectly. I can focus, I can type, I get options shown in the dropdown. It tests the website UI that I expect to interact with as a user.

Some parts though, I still worry are a little too tightly coupled to the implementation. Like, how Algolia works. We don't have any control over the request, nor the response. Honestly, that's entirely their concern. If they change anything about those things in a version bump of the aloglia plugin, this test will likely break which makes it a little brittle for me.